admin/load.php

```diff
@@ -336,3 +336,88 @@ function perflab_get_module_data( $module_file ) {
 
 	return $module_data;
 }
+
+/**
+ * Initialise Admin Pointer
+ *
+ * Handles the bootstrapping of the admin pointer.
+ * Mainly jQuery code that is self-initialising.
+ *
+ * @since 1.0.0
+ * @return void
+ */
+function perflab_admin_pointer() {
```

Comment:
We should limit where we show this pointer, to not be too aggressive. WordPress core does that with its own admin pointers too.

Based on my above comment, a more suitable action for this callback would be `admin_enqueue_scripts`. Through that hook, we can also access the current `$hook_suffix` in this function. I'd suggest we use that to only show our pointer on the WP admin dashboard (`index.php`) and the plugins page from where the plugin gets activated (`plugins.php`). Right now we show the pointer anywhere, throughout the entire admin, which is not only a bit disrupting, but also increases the chance for conflicts with other plugins operating on certain screens.
```suggestion
function perflab_admin_pointer( $hook_suffix ) {
	if ( ! in_array( $hook_suffix, array( 'index.php', 'plugins.php' ), true ) ) {
		return;
	}

```

Reply:
@felixarntz I've updated this, though I am running into a bit of a strange problem. This works great - and I tested / confirmed that the pointer only shows up on the Dashboard and Plugins page. However, I've added a new function to render the `script` in the footer called: `perflab_render_pointer` - I get a jQuery error on any other page that is not the Dashboard / Plugins page. The reason for this is that we are not calling the `wp_enqueue_script` function on other pages, but the script is still getting rendered. I tried duplicated your code above but it renders nothing if I do. I may need to take a further look into it.

Reply:
Thanks @dainemawer I've moved this hook as suggested by @felixarntz to make sure the script is only executed for this page.